src/circuit/gadget/ecc/chip/mul_fixed/short.rs

```diff
@@ -13,48 +16,59 @@ use pasta_curves::{arithmetic::FieldExt, pallas};
 pub struct Config {
     // Selector used for fixed-base scalar mul with short signed exponent.
     q_mul_fixed_short: Selector,
-    q_scalar_fixed_short: Selector,
+    q_mul_fixed_running_sum: Selector,
+    running_sum_config: RunningSumConfig<
+        pallas::Base,
+        { L_VALUE },
+        { FIXED_BASE_WINDOW_SIZE },
+        { NUM_WINDOWS_SHORT },
+    >,
     super_config: super::Config<NUM_WINDOWS_SHORT>,
 }
 
 impl From<&EccConfig> for Config {
     fn from(config: &EccConfig) -> Self {
         Self {
             q_mul_fixed_short: config.q_mul_fixed_short,
-            q_scalar_fixed_short: config.q_scalar_fixed_short,
+            q_mul_fixed_running_sum: config.q_mul_fixed_running_sum,
+            running_sum_config: config.running_sum_short_config.clone(),
             super_config: config.into(),
         }
     }
 }
 
 impl Config {
-    // We reuse the constraints in the `mul_fixed` gate so exclude them here.
-    // Here, we add some new constraints specific to the short signed case.
     pub(crate) fn create_gate(&self, meta: &mut ConstraintSystem<pallas::Base>) {
-        // Check that sign is either 1 or -1.
-        // Check that last window is either 0 or 1.
-        meta.create_gate("Check sign and last window", |meta| {
-            let q_scalar_fixed_short = meta.query_selector(self.q_scalar_fixed_short);
-            let last_window = meta.query_advice(self.super_config.window, Rotation::prev());
-            let sign = meta.query_advice(self.super_config.window, Rotation::cur());
+        // Check that each window uses the correct y_p and interpolated x_p.
+        meta.create_gate("Coordinates check", |meta| {
+            let q_mul_fixed_running_sum = meta.query_selector(self.q_mul_fixed_running_sum);
 
-            let one = Expression::Constant(pallas::Base::one());
+            let z_cur = meta.query_advice(self.super_config.window, Rotation::cur());
+            let z_next = meta.query_advice(self.super_config.window, Rotation::next());
 
-            let last_window_check = last_window.clone() * (one.clone() - last_window);
-            let sign_check = sign.clone() * sign - one;
+            //    z_{i+1} = (z_i - a_i) / 2^3
+            // => a_i = z_i - z_{i+1} * 2^3
+            let word = z_cur - z_next * pallas::Base::from_u64(constants::H as u64);
 
-            vec![
-                q_scalar_fixed_short.clone() * last_window_check,
-                q_scalar_fixed_short * sign_check,
-            ]
+            self.super_config
+                .coords_check(meta, q_mul_fixed_running_sum, word)
```

Comment:
This feels like it's probably a duplicated gate, given that it only queries columns directly from `super_config`.

Reply:
This gate was indeed duplicated across `mul_fixed::short` and `mul_fixed::base_field_elem`. It has now been moved into `mul_fixed.rs`.